lib/ace/editor.js

```diff
@@ -966,7 +966,7 @@ Editor.$uid = 0;
             var lines = text.split(/\r\n|\r|\n/);
             var ranges = this.selection.rangeList.ranges;
     
-            if (lines.length > ranges.length || lines.length < 2 || !lines[1])
+            if (lines.length != ranges.length)
```

Comment:
the code here was trying to allow pasting a full line with ending \n in two places.
Would
```js 
var isFullLine = lines.length == 2 && (!lines[0] || !lines[1])
if (lines.length != ranges.length || isFullLine)
```

work here, or do you think this could still cause problems?

Reply:
Oh, that's true. I now understand why there's a `!lines[1]` right there.

I see it fine. Now this works fine in the cases we had some problems. I don't know other cases but I think your code now makes more sense.

Reply:
Thanks! i've merged the pull request with the above modification, please report if you find any errors.